### PR TITLE
[docs] - Add section to testing page on invoking assets

### PR DESCRIPTION
**Pull request by dpeng817**
Title. Pretty straightforward, I think the interesting bit here are the sections on resources, since those function a bit differently for assets than ops.

### CODE REVIEW

File: docs/content/concepts/testing.mdx [hunk 1]
```diff
@@ -193,13 +193,105 @@ def my_foo_resource(context):
...
+    context = build_op_context(resources={"foo": my_foo_resource.configured({"my_str": "bar"})})
     assert op_requires_foo(context) == "found bar"
 ```
 
-### Testing job execution with configs
+Note that when directly invoking ops, io managers specified on inputs and outputs are not used.
+
+### Testing assets
+
+Similar to ops, <PyObject object="asset" decorator /> decorated functions can be directly invoked. Doing so will invoke the underlying op computation.
```
Comment (erinkcochran87) [inline docs/content/concepts/testing.mdx:204]:
```suggestion
Similar to ops, <PyObject object="asset" decorator /> decorated functions can be directly invoked. Doing so invokes the underlying op computation.
```

Suggesting we use active voice here.

File: docs/content/concepts/testing.mdx [hunk 2]
```diff
@@ -193,13 +193,105 @@ def my_foo_resource(context):
...
+def asset_with_inputs(x, y):
+    return x + y
+
+
+# An example unit test for asset_with_inputs.
+def test_asset_with_inputs():
+    assert asset_with_inputs(5, 6) == 11
+```
+
+Any resources defined on your asset (either directly, or using <PyObject object="with_resources" />) will be picked up when invoking the asset. Simply provide a context using <PyObject object="build_op_context" />.
```
Comment (erinkcochran87) [inline docs/content/concepts/testing.mdx:238]:
```suggestion
Any resources defined on your asset - either directly or using <PyObject object="with_resources" /> - will be picked up when invoking the asset. Simply provide a context using <PyObject object="build_op_context" />.
```

Is using the context required to pick up defined assets? If so, I suggest making this sentence reflect that - right now it sounds like a suggestion.

File: docs/content/concepts/testing.mdx [hunk 3]
```diff
@@ -193,13 +193,105 @@ def my_foo_resource(context):
...
+
+
+def test_asset_with_service():
+    # When invoking asset_with_service, service resource will
+    # automatically be used.
+    result = asset_with_service(build_op_context())
+    ...
+```
+
+Resources provided on <PyObject object="build_op_context" /> will override any resources provided directly on the asset. This means you can avoid using your prod resources when testing.
```
Comment (erinkcochran87) [inline docs/content/concepts/testing.mdx:266]:
```suggestion
Resources provided on <PyObject object="build_op_context" /> will override any resources provided directly on the asset. This means you can avoid using production resources when testing.
```

Other places throughout the docs use `production` - just trying to keep things consistent.

**Review by erinkcochran87 — COMMENTED** [review]
Overall, this looks good to me! I'll leave it to someone else to check for technical accuracy, but I had a few comments about style/grammar.

File: docs/content/concepts/testing.mdx (hunk 2 above)
Comment (dpeng817) [reply docs/content/concepts/testing.mdx:238]: good point - rewording it a bit

File: docs/content/concepts/testing.mdx [hunk 4]
```diff
@@ -199,7 +200,178 @@ def test_op_resource_def():
     assert op_requires_foo(context) == "found bar"
 ```
 
-### Testing job execution with configs
+Note that when directly invoking ops, io managers specified on inputs and outputs are not used.
+
+### Testing assets
+
+Similar to ops, <PyObject object="asset" decorator /> decorated functions can be directly invoked. Doing so invokes the underlying op computation.
```
Comment (sryza) [inline docs/content/concepts/testing.mdx:207]:
```suggestion
Similar to ops, <PyObject object="asset" decorator />-decorated functions can be directly invoked. Doing so invokes the underlying op computation.
```

File: docs/content/concepts/testing.mdx [hunk 5]
```diff
@@ -199,7 +200,178 @@ def test_op_resource_def():
...
+def basic_asset():
+    return 5
+
+
+# An example unit test for basic_asset.
+def test_basic_asset():
+    assert basic_asset() == 5
+```
+
+An asset with input dependencies:
```
Comment (sryza) [inline docs/content/concepts/testing.mdx:225]:
```suggestion
An asset with dependencies:
```

File: docs/content/concepts/testing.mdx [hunk 6]
```diff
@@ -199,7 +200,178 @@ def test_op_resource_def():
     assert op_requires_foo(context) == "found bar"
 ```
 
-### Testing job execution with configs
+Note that when directly invoking ops, io managers specified on inputs and outputs are not used.
+
+### Testing assets
```
Comment (sryza) [inline docs/content/concepts/testing.mdx:205]:
```suggestion
### Testing software-defined assets
```

File: docs/content/concepts/testing.mdx [hunk 7]
```diff
@@ -199,7 +200,178 @@ def test_op_resource_def():
...
+
+
+def test_asset_with_service():
+    # When invoking asset_with_service, service resource will
+    # automatically be used.
+    result = asset_with_service(build_op_context())
+    ...
+```
+
+Resources provided on <PyObject object="build_op_context" /> will override any resources provided directly on the asset. This means you can avoid using production resources when testing.
```
Comment (sryza) [inline docs/content/concepts/testing.mdx:269]: Did we decide to change this behavior?

File: docs/content/concepts/testing.mdx [hunk 8]
```diff
@@ -199,7 +200,178 @@ def test_op_resource_def():
...
+    @resource
+    def mock_resource():
+        return "i am mock"
+
+    # When testing, we override prod_resource with mock_resource.
+    result = asset_uses_service(build_op_context(resources={"service": mock_resource}))
+    ...
+```
+
+### Testing multiple assets together
```
Comment (sryza) [inline docs/content/concepts/testing.mdx:297]:
```suggestion
### Testing multiple software-defined assets together
```

**Review by erinkcochran87 — APPROVED** [review]
One copy change, but otherwise good to go!

**Pull request merged**